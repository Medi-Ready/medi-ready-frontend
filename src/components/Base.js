import React from "react";
import styled from "styled-components";

export const PageTitle = ({ children, className }) => {
  return (
    <PageTitleWrapper className={className}>{children}</PageTitleWrapper>
  );
};

export const PageContent = ({ children, className }) => {
  return (
    <PageContentWrapper className={className}>{children}</PageContentWrapper>
  );
};

const PageTitleWrapper = styled.h2`
  font-size: 20px;
  font-weight: 600;
`;

const PageContentWrapper = styled.article`
  position: relative;
  height: calc(100% - 115px);
  padding: ${({ theme }) => theme.padding.big};
`;
