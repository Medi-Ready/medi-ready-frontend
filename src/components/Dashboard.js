import React from "react";

const Dashboard = () => {
  return (
    <>
      <h2>Dashboard</h2>
    </>
  );
};

export default Dashboard;
