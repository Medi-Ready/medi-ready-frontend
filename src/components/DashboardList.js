import React from "react";

import PropTypes from "prop-types";
import styled from "styled-components";

const Dashboard = ({ src, color, number, text }) => {
  return (
    <Wrapper src={src} color={color}>
      <div>
        <em>{number}</em>
        <span>{text}</span>
      </div>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: inline-block;
  padding: 30px 10px;

  div {
    position: relative;
    min-width: 250px;
    padding: 35px 20px 35px 78px;
    border-radius: 10px;
    background-color: #fff;
    box-shadow: rgb(149 157 165 / 20%) 0px 8px 24px;
    box-sizing: border-box;
  }

  div:before {
    content: "";
    display: block;
    position: absolute;
    border-radius: 50%;
    top: calc(50% - 23px);
    left: 18px;
    padding: 24px;
    background: ${(props) => props.color} url(${(props) => props.src}) no-repeat center;
  }

  div > * {
    display: block;
  }

  em {
    font-size: 18px;
    font-weight: 600;
  }

  span {
    margin-top: 10px;
    font-size: 14px;
    color: #B9B9B9;
  }
`;

Dashboard.propTypes = {
  src: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
};

export default Dashboard;
