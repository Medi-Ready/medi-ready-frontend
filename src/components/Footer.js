import React from "react";
import styled from "styled-components";

const Footer = () => {
  return (
    <StyledFooter>2021 Â© Medi-Ready - medi-ready.com</StyledFooter>
  );
};

const StyledFooter = styled.footer`
  padding: 23px;
  border-top: 1px solid #E0E1EA;
  font-size: 13px;
  color: #B9B9B9;
`;

export default Footer;
