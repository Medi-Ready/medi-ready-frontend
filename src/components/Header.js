import React from "react";
import styled from "styled-components";

const Header = () => {
  return (
    <StyledHeader>
      <div>
        <span className="text">Welcome, 유저이름</span>
        <span className="img">
          <img
            alt="유저이름"
            src="https://avatars.githubusercontent.com/u/60248910?v=4"
          />
        </span>
      </div>
    </StyledHeader>
  );
};

const StyledHeader = styled.header`
  display: flex;
  height: 55px;
  align-items: center;
  background-color: ${({ theme }) => theme.color.white};

  div {
    display: flex;
    margin-left: auto;
    align-items: center;
  }

  .text {
    margin-right: 10px;
  }

  .img {
    width: 40px;
    height: 40px;
  }
`;

export default Header;
