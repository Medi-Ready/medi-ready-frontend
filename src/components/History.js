import styled from "styled-components";
import { useQuery } from "react-query";
import React, { useContext, useEffect, useState } from "react";

import { getPrescriptions } from "../api";
import { ModalContext } from "../contexts/ModalContext";

import Detail from "./Detail";
import Pagination from "./Pagination";
import Loading from "./Shared/Loading";
import HistoryListItem from "./Shared/HistoryListItem";

const History = ({ queryClient }) => {
  const { handleModal } = useContext(ModalContext);
  const [page, setPage] = useState(0);

  const { data, isPreviousData, isFetching, isStale } = useQuery(["prescriptions", page],
    () => getPrescriptions(page), { keepPreviousData: true, staleTime: 5 * 1000 },
  );

  useEffect(() => {
    if (data?.hasMoreData && isStale) {
      queryClient.prefetchQuery(["prescriptions", page + 1], () => getPrescriptions(page + 1));
    }
  }, [data, page, queryClient]);

  const ViewDetail = (prescription) => {
    handleModal(<Detail prescription={prescription} />);
  };

  return (
    <>
      <h2>Prescription History</h2>

      <Wrapper>
        <ul>
          <li className="name-card">
            <b>Name</b>
            <span>Status</span>
            <span>Date</span>
          </li>
        </ul>
        <ul className="list-items">
          {isFetching ? (
            <Loading />
          ) : (
            data.prescriptions.map((prescription) => {
              return (
                <HistoryListItem
                  key={prescription.prescription_id}
                  prescription={prescription}
                  openModal={() => ViewDetail(prescription)}
                />
              );
            })
          )}
        </ul>
      </Wrapper>
      <Pagination
        page={page}
        setPage={setPage}
        hasMoreData={data?.hasMoreData}
        isPreviousData={isPreviousData}
      />
    </>
  );
};

const Wrapper = styled.div`
  margin-top: 20px;
  min-height: 60vh;
  padding: ${({ theme }) => theme.padding.default};
  border-radius: 10px;
  background-color: ${({ theme }) => theme.color.white};
  box-shadow: rgb(149 157 165 / 20%) 0px 8px 24px;

  ul {
    padding-top: 5px;
  }

  ul:first-child {
    border-bottom: 1px solid #e9e9e9;
  }

  li {
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-bottom: 1px solid #eee;
    padding: 30px 15px;
    height: 50px;
  }

  li.name-card b {
    width: 20%;
    font-size: 16px;
    color: #767676;
  }

  li.name-card span {
    color: #767676;
    font-size: 16px;
  }

  li span:nth-child(2) {
    width: 50%;
  }

  li span:nth-child(3) {
    width: 15%;
  }

  li:not(.name-card):hover {
    cursor: pointer;
  }
`;

export default History;
