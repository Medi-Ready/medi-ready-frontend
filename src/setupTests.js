import React from "react";
import ReactDOM from "react-dom";
import "@testing-library/jest-dom";

import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { QueryClient, QueryClientProvider } from "react-query";

import theme from "../src/styles/theme";
import GlobalStyles from "../src/styles/";
import { ModalProvider } from "../src/contexts/ModalContext";

const queryClient = new QueryClient();

export const history = createMemoryHistory({
  initialEntries: ["/"],
});

export const customRender = (children) => (
  render((
    <Router history={history}>
      <ThemeProvider theme={theme}>
        <QueryClientProvider client={queryClient}>
          <GlobalStyles />
          <ModalProvider>
            {children}
          </ModalProvider>
        </QueryClientProvider>
      </ThemeProvider>
    </Router>
  ))
);

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element) => element);
});
