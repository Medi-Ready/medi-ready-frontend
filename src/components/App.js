import React, { useState, useEffect } from "react";
import { Route, Switch, Redirect, useHistory } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "react-query";
import styled, { ThemeProvider } from "styled-components";
import { ReactQueryDevtools } from "react-query/devtools";

import GlobalStyles from "../styles";
import theme from "../styles/theme";
import Header from "./Header";
import Footer from "./Footer";
import Navigation from "./Navigation";
import Login from "./Login";
import History from "./History";
import Settings from "./Settings";
import DashboardList from "./DashboardList";
import Prescription from "../components/Prescription";
import { ModalProvider } from "../contexts/ModalContext";
import { getAuthCheck } from "../api";

const queryClient = new QueryClient();

const App = () => {
  const [user, setUser] = useState("");
  const history = useHistory();

  useEffect(() => {
    const checkLoginStatus = async () => {
      const { result, data } = await getAuthCheck();

      if (result === "success") {
        setUser(data);
      } else {
        setUser("");
        history.push("/login");
      }
    };

    checkLoginStatus();
  }, []);

  return (
    <Wrapper>
      <ThemeProvider theme={theme}>
        <QueryClientProvider client={queryClient}>
          <GlobalStyles />
          <Navigation isLoggedIn={user} onLogout={setUser} />
          <Section>
            <Header />
            <Article>
              <ModalProvider>
                <Switch>
                  <Route path="/dashboard">
                    <DashboardList />
                  </Route>
                  <Route path="/prescription">
                    <Prescription />
                  </Route>
                  <Route path="/prescriptions">
                    <History />
                  </Route>
                  <Route path="/prescriptions/:id">

                  </Route>
                  <Route path="/settings">
                    <Settings />
                  </Route>
                  <Route path="/login">
                    <Login onSuccess={setUser} />
                  </Route>
                  <Route path="/" exact>
                    <Redirect to="/dashboard" />
                  </Route>
                </Switch>
              </ModalProvider>
            </Article>
            <Footer />
          </Section>
        </QueryClientProvider>
      </ThemeProvider>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
`;

const Section = styled.section`
  width: calc(100% - 230px);
  min-height: 100vh;
  background-color: ${({ theme }) => theme.color.grey};
`;

const Article = styled.article`
  height: calc(100% - 115px);
  padding: ${({ theme }) => theme.padding.big};
`;

export default App;
