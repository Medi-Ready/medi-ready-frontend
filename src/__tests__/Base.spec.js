import React from "react";

import { PageTitle, PageContent } from "../components/Base";
import { customRender } from "../setupTests";

describe("base component", () => {
  it("matches snapshot", () => {
    const { container } = customRender(
      <PageTitle>test</PageTitle>,
      <PageContent>test</PageContent>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should show title text", () => {
    const { getByText } = customRender(
      <PageTitle>test</PageTitle>,
    );

    expect(getByText("test")).toBeInTheDocument();
  });

  it("should show content text", () => {
    const { getByText } = customRender(
      <PageContent>test</PageContent>,
    );

    expect(getByText("test")).toBeInTheDocument();
  });
});
