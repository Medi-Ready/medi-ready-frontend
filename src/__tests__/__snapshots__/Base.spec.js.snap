// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base component matches snapshot 1`] = `
<div>
  <h2
    class="sc-hKgJUU jkDiOc"
  >
    test
  </h2>
</div>
`;
